src/Component/QueryEditor.tsx

```diff
@@ -29,11 +28,9 @@ export const QueryEditor: ComponentType<Props> = ({ datasource, onChange, onRunQ
       datasource.metricFindQuery({ query: '', format: 'string' }, undefined).then(
         (result) => {
           const metrics = result.map((value) => ({ label: value.text, value: value.value }));
-
-          const foundMetric = find(metrics, (metric) => metric.value === query.target);
-
-          setMetric(foundMetric === undefined ? { label: '', value: '' } : foundMetric);
-
+          if (typeof query.target === 'string') {
```

Comment:
The previous behaviour did not set the metric when it was not found in `metrics`.

I don't use the UI so am not sure what's better 🤔 

If you'd prefer to change it, I don't mind I guess. But then I'd `return.map()` output from the function straightaway and set metric before that